### PR TITLE
Display course update scheduled message in a notification component

### LINKED ISSUE #1045: 'needs update' action should give info about what to expect
UX feedback: 
> In the interface, it is not clear how to retrieve data from past programs. I
> created a new program for the Edit-a-thon on Nov 2, and clicked on "Schedule
> data update" and the message I got is "update scheduled" but it doesn't say
> anything about when the data will be available, or that you already
> scheduled an update when you click the button the second time. In the demo,
> you mentioned that within 24 hours organizers could retrieve the data. Can
> we add this in the message that pops up?

We can solve this by creating a notification with a relevant explanation.

The trigger for this is in `app/assets/javascripts/components/overview/available_actions.jsx`.

An example of creating a notification via `NotificationActions.addNotification` can be found in `app/assets/javascripts/components/students/enroll_button.jsx`.

### PR DESCRIPTION
**Pull request by arun1595**
Fixes #1045

### CODE REVIEW

File: app/assets/javascripts/utils/api.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { capitalize } from './strings';
+import NotificationActions from '../actions/notification_actions.js';
```
Comment (ragesoss) [inline app/assets/javascripts/utils/api.js]: If we're going to couple this to the response from the server — which is a good idea, although not what I was imagining — then we should not doing this in API.js. As much as possible, API.js should be strictly about querying th endpoint and returning the data.

Adding the Notification could be done by adding handling for the NEEDS_UPDATE with the notification store. ServerActions.needsUpdate() already sends out the 'NEEDS_UPDATE' signal when this action succeeds. Following the patterns of this app would mean adding the 'NEEDS_UPDATE' case to the NotificationStore.

Comment (arun1595) [inline app/assets/javascripts/utils/api.js]: > As much as possible, API.js should be strictly about querying th endpoint and returning the data

Agree. That is why I was initially working on adding this to `available_actions.jsx` where it make it sense to include a notification component.

> Following the patterns of this app would mean adding the 'NEEDS_UPDATE' case to the NotificationStore.

Ah! Makes sense. Let me make the changes and update the PR.

**Comment by arun1595** [comment]
@ragesoss The PR is ready for review. The animation is needed for UX, without it, the user will have no idea about the notification message until he scrolls all the way to the top.

File: app/assets/javascripts/utils/api.js [hunk 2]
```diff
@@ -503,15 +503,12 @@ slide_id=${opts.slide_id}`,
...
       $.ajax({
         type: 'GET',
-        url: `/courses/${courseId}/needs_update.json`,
-        success(data) {
-          alert(data.result);
-          return res(data);
-        }
+        url: `/courses/${courseId}/needs_update.json`
       })
+      .done(obj => res(obj))
```
Comment (ragesoss) [inline app/assets/javascripts/utils/api.js]: You can do this in the 'success' call to match the pattern through most of api.js.

File: app/assets/javascripts/stores/notification_store.js [hunk 3]
```diff
@@ -87,6 +87,17 @@ const NotificationStore = Flux.createStore(storeMethods, (payload) => {
...
+      addNotification({
+        message: payload.data.result,
+        closable: true,
+        type: 'success'
+      });
+
+      $('html, body').animate({scrollTop: 0}, 200)
+        .promise()
+        .then(() => NotificationStore.emitChange());
+      break;
```
Comment (ragesoss) [inline app/assets/javascripts/stores/notification_store.js]: Although this makes sense on it's own in the current UI, this should be addressed by fixing the general problem of notifications appearing when the visible page doesn't include the Notifications section: #234. I think we'll work on a fix for that one soon, so let's cut out the animation here.

Comment (arun1595) [inline app/assets/javascripts/stores/notification_store.js]: Okay.

File: app/assets/javascripts/utils/api.js (hunk 2 above)
Comment (arun1595) [inline app/assets/javascripts/utils/api.js]: Done.

**Pull request merged**